sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+"""
+This module contains functions that will segment images for granule detection.
+"""
+import logging
+
+import numpy as np
+import scipy.ndimage as sndi
+import skimage
+from sklearn.cluster import KMeans
+
+import sunpy
+import sunpy.map
+
+__all__ = [
+    "segment",
+    "get_threshold",
+    "trim_intergranules",
+    "mark_faculae",
+    "kmeans_segment",
+    "cross_correlation",
+]
+
+
+def segment(data_map, skimage_method, res, mark_dim_centers=False):
+    """
+    Segment optical image of the solar photosphere into tri-value maps
+    with 0 = intergranule, 0.5 = faculae, 1 = granule.
+
+    Parameters
+    ----------
+    data_map : `SunPy map`
+        SunPy map containing data to segment
+    skimage_method : `str`

```

Comment:
```suggestion
    skimage_method : {"otsu", "li", "isodata", "mean", "minimum", "yen", "triangle"}
```

This follows the numpydoc style guide for fixed values.